### ISSUE TITLE
Missing assert in test_requests.py test_header_validation

### ISSUE THREAD

**Issue by Sjord** [psf/requests#6551]
[requests/tests/test\_requests.py at main · psf/requests](https://github.com/psf/requests/blob/main/tests/test_requests.py#L1705)

```
    def test_header_validation(self, httpbin):
        """Ensure prepare_headers regex isn't flagging valid header contents."""
        valid_headers = {
            "foo": "bar baz qux",
            "bar": b"fbbq",
            "baz": "",
            "qux": "1",
        }
        r = requests.get(httpbin("get"), headers=valid_headers)
        for key in valid_headers.keys():
            valid_headers[key] == r.request.headers[key]
```

`valid_headers[key] == r.request.headers[key]` should probably be `assert valid_headers[key] == r.request.headers[key]`.

**Comment by swims-hjkl** [comment]
I would like to contribute to solve this issue. @sigmavirus24  can I take this up?

**Comment by nateprewitt** [comment]
Resolved in #6552
